rclcpp/include/rclcpp/executors/multi_threaded_executor.hpp

```diff
@@ -79,7 +79,7 @@ class MultiThreadedExecutor : public executor::Executor
   bool yield_before_execute_;
 
   std::mutex scheduled_timers_mutex_;
-  std::set<TimerBase::SharedPtr> scheduled_timers_;
+  std::set<Waitable::SharedPtr> scheduled_timers_;
```

Comment:
Why change this? They may be `Waitable`, but `TimerBase` is more accurate, e.g. you don't want a subscription to get added here accidentally, I think.

Reply:
This is similar to collapsing the vectors in `AllocatorMemoryStrategy` specific to `SubscriptionBase`, `ClientBase`, `ServiceBase`, etc. to one vector of `Waitable`s, `AnyExecutable` effectively can only contain `Waitable`s, no `TimerBase`s. `AnyExecutable::has_timer()` is also checked before being added to this set.

Also, shouldn't this set be able to include a `Waitable` that has a timer, even if the `Waitable` is not a `TimerBase`?

Reply:
I think the argument here is that if the name is `scheduled_timers_` the expectation is that it should only contain `TimerBase` objects, thus keeping the previous type to enforce that expectation. If the context of how this set used is changing to include all types of Waitables then the name should change to reflect the new context, if not then it should be fine keeping it the more concrete type.

Reply:
This change will be omitted from this PR.